### PR TITLE
Chunk Get POC 2 to solicit input. Do not check-into main.

### PR DESCRIPTION
**Pull request by richkong88**
New POC of Chunk Get with changes based on learnings of first POC.
- No longer use sender/receiver buffer for ChunkGet/ChunkResponse messages. Use scratch space instead. We cannot use the sender/receive buffers because the try_get_xxx_request functions use them while the ChunkGet/ChunkResponse are occuring underneath.
- Because of the way the try_get_xxx_request functions are structured, I needed to move the large response handler out of the main error handler to force the try_get_xxx_request functions to do the same checks in the large response case as in the normal response case.
- The send/receive functions used by ChunkGet/ChunkResponse cannot write the the libspdm_context::last_spdm_request, because the code higher up in try_get_xxx_request uses this variable and assumes it hasn't changed (no hidden requests sent underneath the hood). Hence I duplicated the libspdm_send_request/libspdm_receive_response (added chunk to the name) for now to not modify the variable. I think we can refactor this to just a parameter in the existing versions to not write the last_spdm_request in chunk call.
- Added a unit tests for both requester and responder. Need more for the requester. Requester only unit tests actually are pretty challenging to write because one request can spawn multiple chunks. 

Signed-off-by: Kong, Richard <richard.kong@intel.com>

### CODE REVIEW

File: library/spdm_requester_lib/libspdm_req_end_session.c [hunk 1]
```diff
@@ -104,6 +104,25 @@ libspdm_return_t libspdm_try_send_receive_end_session(libspdm_context_t *spdm_co
         status = LIBSPDM_STATUS_INVALID_MSG_SIZE;
         goto receive_done;
     }
+
+    if (spdm_response->header.request_response_code == SPDM_ERROR
```
[the remark this answers is not in the document]
Comment (richkong88) [reply library/spdm_requester_lib/libspdm_req_end_session.c:108]: Yeah, I think I was just broadly adding the code to anything that called the main error handler.  I'll check on all the response types and only add large message error handlers on any response type that is larger than the MinDataTransferSize (44 bytes)for SPDM 1.2 (44 bytes).

[the remark this answers is not in the document]
Comment (jyao1) [reply library/spdm_requester_lib/libspdm_req_end_session.c:108]: There is no large error message.
Error message must NOT use chunking. See paragraph 701.

[the remark this answers is not in the document]
Comment (richkong88) [reply library/spdm_requester_lib/libspdm_req_end_session.c:108]: I meant I'll only add a handler for the case of ERROR response with error code == LARGE_RESPONSE, if the expected response type (e.g. spdm_certificates_response_t) type has a size > MinDataTransferSize (44 bytes).

[the remark this answers is not in the document]
Comment (jyao1) [reply library/spdm_requester_lib/libspdm_req_end_session.c:108]: I see.

File: include/internal/libspdm_common_lib.h [hunk 2]
```diff
@@ -422,6 +444,9 @@ void libspdm_internal_dump_data(const uint8_t *data, size_t size);
  **/
 void libspdm_internal_dump_hex(const uint8_t *data, size_t size);
 
+
+void libspdm_trace_message(void* message, size_t message_size);
```
Comment (jyao1) [inline include/internal/libspdm_common_lib.h:448]: anyone using it?

Comment (richkong88) [reply include/internal/libspdm_common_lib.h:448]: I was using it in a local build. Did not plan on checking this in at this time and did not mean to include it in the pull request.

File: library/spdm_requester_lib/libspdm_req_end_session.c (hunk 1 above)
[the remark this answers is not in the document]
Comment (richkong88) [reply library/spdm_requester_lib/libspdm_req_end_session.c:108]: Good idea. Let me see what I can do.

**Comment by richkong88** [comment]
Closing as this was just a POC to get feedback.

**Pull request closed without merging**